### DISCUSSION TITLE
百度网盘非官方接口通过webdav无法播放视频并提示403

### DISCUSSION THREAD

**Discussion by zhangxiongbiao** [AlistGo/alist#4527] [Q&A]
### Please make sure of the following things

- [X] I have read the [documentation](https://alist.nn.ci).
- [X] I'm sure there are no duplicate issues or discussions.
- [X] I'm sure it's due to `alist` and not something else(such as `Dependencies` or `Operational`).
- [X] I'm sure I'm using the latest version

### Alist Version / Alist 版本

v3.18.0

### Driver used / 使用的存储驱动

Baidu_netdisk

### Describe the bug / 问题描述

百度网盘中的非官方接口无法正常使用，现在通过该接口访问百度中的视频时，直接报错提示403。当我把版本降级到3个月前的低版本时发现可以正常使用。我翻了下 Release 日志，看到可能是 [v3.13.1](https://github.com/alist-org/alist/releases/tag/v3.13.1) 版本中的 https://github.com/alist-org/alist/pull/3652 这个提交造成的。

### Reproduction / 复现链接

https://github.com/alist-org/alist/pull/3652

**Comment by welcome** [comment]
Thanks for opening your first issue here! Be sure to follow the issue template!

**Comment by anwen-anyi** [comment]
是这样的   常规的那样方法不行了，需要  `带上Range头`

**Comment by zhangxiongbiao** [comment]
> 是这样的 常规的那样方法不行了，需要 `带上Range头`

怎么带？需要在客户端上做配置吗？客户端用的 nplayer ...

**Reply by anwen-anyi** [reply]
不太清楚 还没用过这个`非官方`接口没搞懂目前，帮你 @Admirepowered 一下大佬顺便学习学习

**Comment by xhofe** [comment]
不保证`非官方接口`的可用性，有关`非官方接口`的问题可以在discussion中讨论。

**Comment by Admirepowered** [comment]
使用浏览器或者其他播放器会自动带Range，你看你是不是UA没改为netdisk，而不是官方的pan.baidu.com

**Reply by anwen-anyi** [reply]
好像不太行？
![image](https://github.com/alist-org/alist/assets/56105412/d842cdc2-22a7-44f5-a004-0252195ff220)

**Reply by Admirepowered** [reply]
我刚测试了一下，官方已经回退了，你懂的，百度一直在修改，所以我也不知道什么东西能用，可以回退这个pr了，origin=dlna又不会hitcode118了。

**Reply by anwen-anyi** [reply]
哈哈 我当时想的是两个不同的非官方同时存在~~ （你在pr之后的几天）
不过我自己尝试修改并存了一下但是没改对（非专业开发人员）

之前的哪个非官方播放视频倒是还可以，就是不能下载东西了

**Reply by Admirepowered** [reply]
是这样的，百度改来改去很烦人，并存的话需要判断文件类型，你得问问其他开发者了，这两个用法其实算一个接口，origin=dlna指定的是文件类型为流媒体，傻逼百度的服务器不按RFC标准来，所以会403

**Reply by anwen-anyi** [reply]
OK原来是这样子...那确实不干人事百度  哈哈
之前的那个版本开始提交的时候pr的大佬也说了只是用来看视频的，但是发现能下载文件（意料之外的）

**Comment by anwen-anyi** [comment]
刚刚测试了一下 遇到了个奇葩的事儿...
大佬的pr https://github.com/alist-org/alist/commit/53a1c42 是合并在 3.13.1版本发布的
![image](https://github.com/alist-org/alist/assets/56105412/4f851c6c-f309-4cea-aaa3-8013e68b62bd)

-----

然后我就去下载了一个 [V3.12.2](https://github.com/alist-org/alist/releases/tag/v3.12.2)的版本，之前的时候他们就说这个版本可以播放，在上个月的时候我记得确实是可以播放的 也有好多Tg群友说可以播放
我刚刚测试去用的时候发现也不能播放，但是我当时的浏览器UA还是`netdisk`忘记关掉了启动了3.12.2的版本竟然能播放起来了，我以为是卡了，我就又把  `netdisk`的UA 去掉了，咦~不能播放了，我就又双叒叕把 `netdisk`的UA加上竟然又能播放了 哈哈.....


### 复现视频
（直接看可能看不了 可以下载查看）

https://github.com/alist-org/alist/assets/56105412/ad2a3cfa-973c-4890-958f-c6848a03fb2d
